R/add_thresholds_transition_risk.R

```diff
@@ -0,0 +1,130 @@
+#' Adds low and high thresholds for transition risk score
+#'
+#' @param emissions_profile_products A dataframe
+#' @param all_uuids_scenario_sectors A dataframe
+#' @param scenarios A dataframe
+#'
+#' @keywords internal
+#'
+#' @return A dataframe.
+#' @export
+#'
+#' @examples
+#' library(tiltToyData)
+#' library(readr)
+#' library(dplyr)
+#' options(readr.show_col_types = FALSE)
+#'
+#' emissions_profile_products <- read_csv(toy_emissions_profile_products_ecoinvent())
+#' all_uuids_scenario_sectors <- read_csv(toy_sector_profile_companies()) |>
+#'   select(-c("companies_id", "company_name", "clustered")) |>
+#'   distinct()
+#' scenarios <- read_csv(toy_sector_profile_any_scenarios())
+#'
+#' output <- add_thresholds_transition_risk(
+#'   emissions_profile_products,
+#'   all_uuids_scenario_sectors,
+#'   scenarios
+#' )
+#' output
+add_thresholds_transition_risk <- function(emissions_profile_products,
+                                           all_uuids_scenario_sectors,
+                                           scenarios) {
+  check_crucial_cols(emissions_profile_products, c(col_uuid(), col_co2_footprint()))
+  check_crucial_cols(all_uuids_scenario_sectors, c(
+    col_uuid(), col_type(),
+    col_sector(), col_subsector()
+  ))
+  check_crucial_cols(scenarios, c(
+    col_type(), col_sector(), col_subsector(),
+    col_year(), col_scenario(), col_targets()
+  ))
+
+  epa_profile_ranking <- epa_compute_profile_ranking(emissions_profile_products) |>
+    select_crucial_ranking()
+
+  spa_reduction_targets <- spa_compute_profile_ranking(
+    all_uuids_scenario_sectors,
+    scenarios
+  ) |>
+    select_crucial_target()
+
+  full_join(
+    epa_profile_ranking,
+    spa_reduction_targets,
+    by = col_uuid(),
+    relationship = "many-to-many"
+  ) |>
+    add_transition_risk_score(col_ranking(), col_targets()) |>
+    add_benchmark_tr_score(col_ranking(), col_targets()) |>
+    distinct() |>
+    add_low_high_transition_risk_thresholds() |>
+    select(-c(col_scenario(), col_year()))
+}
+
+#' Calulate `transition_risk_score` column
+#'
+#' @param data Dataframe.
+#' @param profile_ranking Dataframe column.
+#' @param reduction_targets Dataframe column.
+#' @keywords internal
+#' @export
+add_transition_risk_score <- function(data,
+                                      col_ranking = col_ranking(),
+                                      col_target = col_targets()) {
+  mutate(
+    data,
+    transition_risk_score = ifelse(
+      is.na(data[[col_ranking]]) | is.na(data[[col_target]]),
+      NA,
+      (data[[col_ranking]] + data[[col_target]]) / 2
```

Comment:
```suggestion
      (.data[[col_ranking]] + .data[[col_target]]) / 2
```

Reply:
In this specific case `data` and `.data` yield the same but `.data` more clearly communicates we're referring to a column in the data-mask.

Also `.data` avoids a bug if later the code does something by groups. I show that in a reprex.

<details>

``` r
library(dplyr, warn.conflicts = FALSE)

data <- tibble(head(mtcars)[1:2])

# same
data |> mutate(n = length(data[["cyl"]]))
#> # A tibble: 6 × 3
#>     mpg   cyl     n
#>   <dbl> <dbl> <int>
#> 1  21       6     6
#> 2  21       6     6
#> 3  22.8     4     6
#> 4  21.4     6     6
#> 5  18.7     8     6
#> 6  18.1     6     6

data |> mutate(n = length(.data[["cyl"]]))
#> # A tibble: 6 × 3
#>     mpg   cyl     n
#>   <dbl> <dbl> <int>
#> 1  21       6     6
#> 2  21       6     6
#> 3  22.8     4     6
#> 4  21.4     6     6
#> 5  18.7     8     6
#> 6  18.1     6     6



# Not same
data |> mutate(n = length(data[["cyl"]]), .by = "mpg")
#> # A tibble: 6 × 3
#>     mpg   cyl     n
#>   <dbl> <dbl> <int>
#> 1  21       6     6
#> 2  21       6     6
#> 3  22.8     4     6
#> 4  21.4     6     6
#> 5  18.7     8     6
#> 6  18.1     6     6

data |> mutate(n = length(.data[["cyl"]]), .by = "mpg")
#> # A tibble: 6 × 3
#>     mpg   cyl     n
#>   <dbl> <dbl> <int>
#> 1  21       6     2
#> 2  21       6     2
#> 3  22.8     4     1
#> 4  21.4     6     1
#> 5  18.7     8     1
#> 6  18.1     6     1
```

<sup>Created on 2024-06-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

</details>